### ISSUE TITLE
[bug] submit errors fail to render in-line

### ISSUE THREAD

**Issue by clphillips** [lookfirst/mui-rff#154]
**Describe the bug**
According to [React Final Form docs](https://final-form.org/docs/react-final-form/examples/submission-errors) returning an object from the submit handler with the shape of the submitted form values should make those errors available to be displayed in-line (similar to validation errors). That does not appear to be the case in version 1.10.2 of mui-rff.

**To Reproduce**

Clone example sandbox: https://codesandbox.io/s/react-final-form-material-ui-example-tqv09

Change `onSubmit` to:

```
const onSubmit = async values => {
  return {
    firstName: 'Bad first name!',
  };
};
```

Example:
https://codesandbox.io/s/react-final-form-material-ui-example-g6p38

**Expected behavior**
`firstName` renders an error on submit similar to leaving `firstName` blank.

**Additional Context**
I've looked through the tests and noticed that none of the tests verify errors are displayed when returned from the submit handler.

**Comment by lookfirst** [comment]
@clphillips Thank you for the report. You are correct that I don't have an actual test for that. I just kind of assumed it would work! My bad! I'll check on this and report soon.

**Comment by lookfirst** [comment]
Hmm... this seems to be working for me...

![image](https://user-images.githubusercontent.com/85355/76014445-8112fa80-5ece-11ea-8740-e717013e5698.png)

**Comment by lookfirst** [comment]
Ok, thanks for the clarification, you're totally right!

Just to explain, I was abusing React for error handling before (calling setState outside of the current component). Latest React now logs a warning/error when you try to do that, so I had to fix it.

I had to wrangle Final Form to do what I want because of the way Field and subscription works...  which took me like a day or two of trial and error to figure out. As you discovered, I had no tests for the onSubmit, so this one slipped through. I'll go back and add the tests and fix this issue. Give me some time to figure it out as I'm sure it will be a headache.

![image](https://user-images.githubusercontent.com/85355/76039176-740eff00-5f00-11ea-9def-9edf5c2d6994.png)
